File: examples/complete-catalog-example/frontend/types/employeeDepartment.ts
/**
 * Employee Department Types - Example using centralized types
 */

// Base types that would normally come from @packages/api-types
export type UUID = string
export type Timestamp = string

export interface BaseCatalog {
  id: UUID
  tenant_id?: UUID
  code?: string
  name: string
  description?: string
  is_active: boolean
  created_at: Timestamp
  updated_at: Timestamp
}

export interface CatalogCreateRequest {
  code?: string
  name: string
  description?: string
  is_active?: boolean
}

export interface CatalogUpdateRequest {
  code?: string
  name?: string
  description?: string
  is_active?: boolean
}

export interface CatalogFilters {
  page?: number
  per_page?: number
  search?: string
}

export interface PaginatedResponse<T> {
  items: T[]
  total: number
  page: number
  per_page: number
  pages: number
}

// Employee Department extends BaseCatalog with tenant isolation
export interface EmployeeDepartment extends BaseCatalog {
  tenant_id: UUID
}

// Request types for API operations
export type EmployeeDepartmentCreate = CatalogCreateRequest & {
  // Additional fields specific to EmployeeDepartment can go here
}

export type EmployeeDepartmentUpdate = CatalogUpdateRequest

// Response type for API operations
export type EmployeeDepartmentResponse = EmployeeDepartment

// Filter type for list operations
export type EmployeeDepartmentFilters = CatalogFilters & {
  // Additional filters specific to EmployeeDepartment
  search?: string
  is_active?: boolean
  name_contains?: string
  code_contains?: string
}

// Paginated response type
export type EmployeeDepartmentPaginatedResponse = PaginatedResponse<EmployeeDepartment>

// This replaces ~30 lines of manual type definitions:
//
// export interface EmployeeDepartment {
//   id: UUID
//   tenant_id: UUID
//   code?: string
//   name: string
//   description?: string
//   is_active: boolean
//   created_at: string
//   updated_at: string
// }
//
// export interface EmployeeDepartmentCreate {
//   code?: string
//   name: string
//   description?: string
//   is_active?: boolean
// }
//
// export interface EmployeeDepartmentUpdate {
//   code?: string
//   name?: string
//   description?: string
//   is_active?: boolean
// }

File: examples/complete-catalog-example/backend/app/schemas/employee_department.py
"""
Employee Department Schemas - Example using schema generator
"""

from app.utils.schema_generator import get_catalog_schemas

# Generate all schemas automatically - no manual definition needed!
EmployeeDepartmentSchemas = get_catalog_schemas("EmployeeDepartment")
EmployeeDepartmentBase = EmployeeDepartmentSchemas["Base"]
EmployeeDepartmentCreate = EmployeeDepartmentSchemas["Create"]
EmployeeDepartmentUpdate = EmployeeDepartmentSchemas["Update"]
EmployeeDepartmentResponse = EmployeeDepartmentSchemas["Response"]

# Export for easy import
__all__ = [
    "EmployeeDepartmentBase",
    "EmployeeDepartmentCreate",
    "EmployeeDepartmentUpdate", 
    "EmployeeDepartmentResponse",
]

# This replaces ~80 lines of manual schema definitions:
#
# class EmployeeDepartmentBase(BaseModel):
#     name: str = Field(..., min_length=1, max_length=100)
#     code: str | None = Field(None, max_length=50)
#     description: str | None = Field(None)
#     is_active: bool = Field(default=True)
#     tenant_id: UUID | None = Field(None)
#
# class EmployeeDepartmentCreate(EmployeeDepartmentBase):
#     pass
#
# class EmployeeDepartmentUpdate(BaseModel):
#     name: str | None = Field(None, min_length=1, max_length=100)
#     code: str | None = Field(None, max_length=50)
#     description: str | None = Field(None)
#     is_active: bool | None = Field(None)
#
# class EmployeeDepartmentResponse(EmployeeDepartmentBase):
#     id: UUID
#     tenant_id: UUID
#     created_at: str
#     updated_at: str
